### ISSUE TITLE
MAME2003 / NEO-GEO Module Loading Issues

### ISSUE THREAD

**Issue by jodriscoll** [KMFDManic/NESC-SNESC-Modifications#54]
### Issue Description:
Attempting to load MAME32 and NEO-GEO titles through RetroArch, I'm experiencing a small lag (black screen), then being sent back to the game select screen (both cores). Tried watching your videos and testing out a plethora of module combinations, _unfortunately_, same issue is persistent. I'm sure I'm missing something; just need a little guidance as the `.hmod` files _do not_ list dependencies clearly.

- **Video Demonstration:** https://www.youtube.com/watch?v=qfd0iXsIvxA

### Path Examples:
- **MAME2003:** ```/bin/mame2003 /usr/share/games/CLV-Z-UIQRV/DonPachi.zip```
- **NEO-GEO:** ```/bin/mame2003 /usr/share/games/CLV-ZXXTXC/Strikers_1945_Plus.zip NEO-GEO```

### CORES and EMULATION:
```bash
_km_bio_neo_mame2003_req
_km_fbalpha2012_neogeo_10_24_17
_km_genesis_plus_gx_10_28_17
_km_handy_12_24_17
_km_mame2003_12_28_17
_km_mednafen_pce_fast_10_23_17
_km_picodrive_32x_11_7_17
_km_retroarch_167_12_25_17
_km_snes9x_12_25_17
_km_vbam_12_21_17
```

#### MODIFICATIONS:
```bash
_usb_tiny7zx_dynamic
remove_thumbnails
MAME_Bios
```
	
#### CHEATS:
```bash
_km_retroarch_cheats_gba_12_20_16
_km_retroarch_cheats_mame2003_10_31_17
_km_retroarch_cheats_md_12_20_16
_km_retroarch_cheats_nes_12_20_16
_km_retroarch_cheats_snes_12_20_16
```

**Comment by KMFDManic** [comment]
Via USB-HOST?

**Comment by KMFDManic** [comment]
You cant rename mame roms. They need to maintain integrity as originally named.  Add as archives, leaving base names intact.

Strikers_1945_Plus.zip completely breaks the ability to read from internal database.  Otherwise, roms need to be correctly aligned with the cores.

**Comment by jodriscoll** [comment]
@KMFDManic – I haven't introduced USB-HOST, however, I plan to do so over the next week (device should be delivered today).

I was unaware of the naming convention being required with the archive file; is there a way to validate I've named it correctly and accordingly? Now that I've renamed them, I'm unsure what to revert their names back too; is there something within the archive file that will define what the archive should be named as?

It's okay to rename the listing of the file within hakchi2 though, right?

**Comment by KMFDManic** [comment]
Best to readd the game, if able to, since youd have to change desktop files, as well as main rom name, otherwise.  But, in Hakchi2, yes...You can name however you'd like.  How many did you rename?

**Comment by ArieKindicator** [comment]
Regarding the Neo Geo core. I noticed that the RetroArch options menu is almost empty for this core. Before I could change a lot of things here.
Quick Menu -> Options. The only available options are: diagnostics, unibios, and overclock.
In previous releases in the options menu I could change, which machine (AES, MVS and regions AES Japan) and which bios to use (basically every available bios in the neogeo.zip file).

It could be very well that I'm doing it wrong this time or different then before or it might be related to usb-host (which I use since mid December 17).

**Comment by jodriscoll** [comment]
@KMFDManic – luckily, I've only renamed a few files to test the MAME2003 functionality. I'll experiment with running the file under it's original (archive (.zip)) file name and report back.

Does this requirement only apply to games that run through the MAME core(s)? I noticed that I'm able to name SNES/SEGA/GBA/GBC/GB/NES/TurboGrafx/etc. to whatever I'd like.

Thanks for responding!

**Comment by pilotS1** [comment]
I dont think to mame but to FBA you have to stick to the default names. I fell into the same trap myself. For Example mslug3.zip should not be added as mslug3(1).zip and tried to be loaded with fba. Wont work.

**Comment by KMFDManic** [comment]
@pilotS1 @jodriscoll @ArieKindicator Yes, Arcade Games should never be renamed.  They need to maintain their "source" name, to properly validate within the database, and pull up the proper roms from within.  As far as NEO-GEO, on USB-HOST, it functions much differently than on Non-USB-HOST.  You can run the Bios HMODS in my Core Set when off of USB-HOST.  But, when on USB-HOST, there are other ways of doing it.  

1.  Make a Dummy Folde within hakchi/games directory, and copy and paste bios+neo geo roms into there, and load as content, like in THIS video.

https://www.youtube.com/watch?v=A3uLD8QSS5g

I will do a more in depth NEO-GEO Video showing other ways to run them.  But, Dummy method is by far the best way for now.

This naming convention only applies to Arcade Games, for FBA and MAME.  ALL other Systems can run with whatever names, except in case of USB-HOST, where special characters need removed.  I posted videos on this, too.  And, I will be posting another.

**Comment by KMFDManic** [comment]
I will be posting a NEO-GEO Video, to help with USB-HOST.  But, your issue, being directly related to incorrectly naming Roms, I am now closing it.  Let me know if you need anymore assistance:)

**Comment by jodriscoll** [comment]
@KMFDManic – the issue was in fact the renaming of the .zip files! I've since resolved the issue with some (if not most) of the arcade titles. However, now that I've got my USB-HOST up and running, I can't seem to get a single NEO-GEO title to run (literally).

I'll reach out if I run into additional issues, but as of right now, the issue that pertains to this instance has been relatively resolved!
